src/App/middlewares/permission.js

```diff
@@ -0,0 +1,13 @@
+import asyncHandler from '../helpers/asyncHandler';
+import ErrorResponse from '../utils/errorResponse';
+import findUser from '../../Database/services/user';
+
+
+export const isAdmin = asyncHandler(async (req, res, next) => {
```

Comment:
Please transform this into a Permissions class and IsAdmin can be a property inside the class.